StudentDriver/StudentDriver/DrivePage.xaml.cs

```diff
@@ -1,24 +1,145 @@
 ﻿using System;
 using System.Collections.Generic;
-
+using System.Threading.Tasks;
+using System.Diagnostics;
+using Plugin.Geolocator;
+using Plugin.Geolocator.Abstractions;
 using Xamarin.Forms;
+using StudentDriver.Services;
+using StudentDriver.Models;
 
 namespace StudentDriver
 {
 	public partial class DrivePage : ContentPage
 	{
-		private bool drivingButtonPressed = false;
-		public DrivePage ()
+		private bool isStudentDriving = false;
+		//speed and distance in miles (imperial)
+		private double currentAverageSpeed = 0.0;
+		private TimeSpan currentTime;
+		private List<DrivePoint> positions = new List<DrivePoint>();
+		private IGeolocator locator;
+
+		protected override void OnAppearing()
 		{
-			InitializeComponent ();
-			NavigationPage.SetHasNavigationBar (this, false);
-			drivingButton.Clicked += async (object sender, EventArgs e) => {
-				drivingButton.BackgroundColor = drivingButtonPressed ? AppColors.Third : AppColors.Fourth;
-				drivingButtonPressed = !drivingButtonPressed;
-				drivingButton.Text = drivingButtonPressed ? "Stop" : "Start";
+			base.OnAppearing();
+
+			try
+			{
+				locator = CrossGeolocator.Current;
+				locator.DesiredAccuracy = 5;
+				locator.PositionChanged += (object s, PositionEventArgs eventArg) =>
+				{
+					var currentPosition = eventArg.Position;
+					if (positions.Count < 100) // every 100 points, we will add to the database.
```

Comment:
Just a round number, I don't want to store too much in memory so I will write every 100 points to the database